### PR TITLE
Added Spring SecurityContext instructions

### PR DESCRIPTION
**Pull request by CharlesDuboisSAP**
## What Has Changed?

Added new instructions to propagate the Spring `SecurityContext` to `@Async` calls.
- Removed a redundant tip
- Fixed grammar
- Added the new instructions in a tip

## Manual Checks?

- [x] Check spelling and grammar, e.g., using Grammarly.
- [x] Verify links still work, e.g., if `id` or the name of a file is changed.
- [x] Update the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js), e.g., if a feature is added.

### CODE REVIEW

File: docs-java/features/multi-tenancy/thread-context.mdx [hunk 1]
```diff
@@ -105,6 +100,16 @@ public class AsynchronousConfiguration implements AsyncConfigurer {
 
 You can read more about the `@Async` functionality [here](https://www.baeldung.com/spring-async).
 
+:::tip
+The Spring `SecurityContext` can be propagated to `@Async` calls.
+Replace the above executor with this one:
+
+```java
+new DelegatingSecurityContextExecutor(ThreadContextExecutors.getExecutor());
```
Comment (Johannes-Schneider) [inline docs-java/features/multi-tenancy/thread-context.mdx:108]: This requires (at least transitively) the some additional Spring security dependency, which is not shipped with the SDK out of the box, right?
If so, we should mention that here more explicitly I think.

Comment (CharlesDuboisSAP) [reply docs-java/features/multi-tenancy/thread-context.mdx:108]: I didn't add any dependency to the archetype. It must be shipped out of the box.

File: docs-java/features/multi-tenancy/thread-context.mdx [hunk 2]
```diff
@@ -79,16 +79,11 @@ Future runningTask = ThreadContextExecutors.submit(myTask);
 Operations that are submitted this way will be executed by a single instance of `ThreadContextExecutorService`.
 By default, this instance is based on a `CachedThreadPool`, which manages the concurrency internally.
 
-:::tip Spring Integration
```
Comment (MatKuhr) [inline docs-java/features/multi-tenancy/thread-context.mdx:82]: why move this?

Comment (CharlesDuboisSAP) [reply docs-java/features/multi-tenancy/thread-context.mdx:82]: I didn't move it I removed it. The content of the tip is the same sentence as the introduction of the next chapter. And the link links to the line below, and therefore is useless.

**Review by Johannes-Schneider — APPROVED** [review]
LGTM

**Pull request merged**